Updated release script documentation and command names (#17929)
facebook/react: scripts/release/README.md

A typical release goes like this:
1. When a commit is pushed to the React repo, [Circle CI](https://circleci.com/gh/facebook/react/) will build all release bundles and run unit tests against both the source code and the built bundles.

> 2. Next the release is [**published to the `next` channel**](#publishing-next) using the [`prepare-next`](#prepare-next) and [`publish`](#publish) scripts. (Currently this process is manual but might be automated in the future using [GitHub "actions"](https://github.com/features/actions).)

Reviewer: ```suggestion
2. The release is then [**published to the `next` channel**](#publishing-next) using the [`prepare-next`](#prepare-next) and [`publish`](#publish) scripts. (Currently this process is manual but might be automated in the future using [GitHub "actions"](https://github.com/features/actions).)
```

When publishing an experimental release, use the `experimental` tag:

```sh
scripts/release/publish.js --tags experimental
```

If the OTP code expires while publishing, re-run this command and answer "y" to the questions about whether it was expected for already published packages.

<sup>1: We have plans to make this less awkward. Ideally these releases will be published by a cron job.</sup>

## Publishing a Stable Release

Stable releases should always be created from the "next" channel. This encourages better testing of the actual release artifacts and reduces the chance of unintended changes accidentally being included in a stable release.

To prepare a stable release, choose a "next" version and run the [`prepare-stable`](#prepare-stable) script <sup>1</sup>:

```sh
scripts/release/prepare-stable.js --version=0.0.0-241c4467e
```

This script will prompt you to select stable version numbers for each of the packages. It will update the package JSON versions (and dependencies) based on the numbers you select.

Once this step is complete, you're ready to publish the release:

```sh

> scripts/release/publish.js --tags latest

Author: Note this is arguably a significant change. If "next" releases are identified as always being 0.0.0-<hash>, and "latest" releases are always semver numbers, it doesn't make sense for us to update `next` to match `latest` on a stable release. I think that's fine though.

#### Example usage
To create a build from the current branch and revision:
```sh
scripts/release/create-next.js
```

## `prepare-next`
Downloads build artifacts from Circle CI in preparation to be published to NPM as a "next" release.

All artifacts built by Circle CI have already been unit-tested (both source and bundles) but these candidates should **always be manually tested** before being published. Upon completion, this script prints manual testing instructions.

#### Example usage
To prepare the artifacts created by [Circle CI build 80592](https://circleci.com/gh/facebook/react/80592#artifacts/containers/0) you would run:
```sh
scripts/release/prepare-next.js --build=80592
```

## `prepare-stable`
Checks out a "next" release from NPM and prepares it to be published as a stable release.

This script prompts for new (stable) release versions for each public package and updates the package contents (both `package.json` and inline version numbers) to match. It also updates inter-package dependencies to account for the new versions.

> Next release have already been tested but it is still a good idea to **manually test and verify a release** before publishing to ensure that e.g. version numbers are correct. Upon completion, this script prints manual testing instructions.

Reviewer: ```suggestion
This release has already been tested, but it is still a good idea to **manually test and verify a release** before publishing to ensure that e.g. version numbers are correct. Upon completion, this script prints manual testing instructions.
```
